backend/de.metas.contracts/src/main/java/de/metas/contracts/invoicecandidate/FlatrateTerm_Handler.java

```diff
@@ -203,7 +203,7 @@ public void setOrderedData(@NonNull final I_C_Invoice_Candidate ic)
 		ic.setQtyEntered(calculateQtyOrdered.toBigDecimal());
 		ic.setC_UOM_ID(calculateQtyOrdered.getUomId().getRepoId());
 
-		final ProductId productId = ProductId.optionalOfRepoId(ic.getM_Product_ID())
+		final ProductId productId = ProductId.optionalOfRepoId(term.getM_Product_ID())
```

Comment:
why?

Reply:
It was a typo introduced here https://github.com/metasfresh/metasfresh/commit/7cb4f03ae9ef1e8041ba9d732edb9cfc1f727249#diff-4d8ffe171e8889a74796e34c9a5456fa5debf755d14f2cc3f57d2e059007ca2bR206  and I fixed it